### DISCUSSION TITLE
cargo cross compile on v21.2 gives "can't find crate for `pest`" error

### DISCUSSION THREAD

**Discussion by bmallya** [vectordotdev/vector#12751] [General]
Hello,
I tried to cross compile vector v21.2 with target as armv7-unknown-linux-gnueabihf, but ended up getting the error on pest. Has anyone come across this?
`make cross-build-armv7-unknown-linux-gnueabihf 
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
make -k cross-image-armv7-unknown-linux-gnueabihf
fatal: not a git repository (or any of the parent directories): .git
make[1]: Entering directory '/home/bharath/vector/vector'
podman build \
	--tag vector-cross-env:armv7-unknown-linux-gnueabihf \
	--file scripts/cross/armv7-unknown-linux-gnueabihf.dockerfile \
	scripts/cross
STEP 1/3: FROM rustembedded/cross:armv7-unknown-linux-gnueabihf
STEP 2/3: COPY bootstrap-ubuntu.sh .
--> Using cache 458a3ed0509f1d8d49a9da8dafe6035128b53bcd9bb1ed51fda969cbfdbe825f
--> 458a3ed0509
STEP 3/3: RUN ./bootstrap-ubuntu.sh
--> Using cache df9a5bd98d213653ba8b99be665b5fa9f84a1d148e0d7a917cb747ac2c3c649d
COMMIT vector-cross-env:armv7-unknown-linux-gnueabihf
--> df9a5bd98d2
Successfully tagged localhost/vector-cross-env:armv7-unknown-linux-gnueabihf
df9a5bd98d213653ba8b99be665b5fa9f84a1d148e0d7a917cb747ac2c3c649d
make[1]: Leaving directory '/home/bharath/vector/vector'
cross build \
	--release \
	--target armv7-unknown-linux-gnueabihf \
	--no-default-features \
	--features target-armv7-unknown-linux-gnueabihf
info: syncing channel updates for '1.58.1-x86_64-unknown-linux-gnu'

  1.58.1-x86_64-unknown-linux-gnu unchanged - rustc 1.58.1 (db9d1b20b 2022-01-20)

info: checking for self-updates
   Compiling pest_meta v2.1.3
error[E0463]: can't find crate for `pest`
  --> /cargo/registry/src/github.com-1ecc6299db9ec823/pest_meta-2.1.3/src/lib.rs:17:1
   |
17 | extern crate pest;
   | ^^^^^^^^^^^^^^^^^^ can't find crate

For more information about this error, try `rustc --explain E0463`.
error: could not compile `pest_meta` due to previous error
make: *** [Makefile:243: cross-build-armv7-unknown-linux-gnueabihf] Error 101
`

**Comment by jszwedko** [comment]
Hi @bmallya !

That's odd. I can't get podman working on my system, but I was able to run the same make command using docker just now on the `v0.21.2` tag. Do you have docker installed? You could try that by `export CONTAINER_TOOL=docker` before running the `make` command.

**Reply by bmallya** [reply]
Thanks for the update @jszwedko. When I run the make (both musl and gnu) I run into the same error even with Docker. If you were able to compile v0.21.2 for armv7-unknown-linux-gnueabihf successfully, would it be possible to pass me the binary? I will try and see if I can use it.

**Reply by jszwedko** [reply]
Hi @bmallya ! We do publish binaries for that target. You can see them here for v0.21.2: https://packages.timber.io/vector/0.21.2/

**Reply by bmallya** [reply]
Hello @jszwedko, we cannot use the published binary because of the issue -  #https://github.com/vectordotdev/vector/issues/11970

**Reply by jszwedko** [reply]
Hi @bmallya . Ah I see. I believe building it locally would have the same issue, no?

**Reply by bmallya** [reply]
The last time I had built v18.0 using a subset of targets and I was able to get it to work. 
I tried to do the same thing with the latest version but seem to hit this issue(which for some reason I had not seen earlier. Was this pest package added recently into vector?)

**Reply by jszwedko** [reply]
Ah, that's interesting, I'm not sure why building it locally would have resulted in any difference. I can run it, anyhow, and share the resulting binary to see if it works for you. Which features did you have enabled?

The `pest` package has been in there a while, since 0.11.0.

**Reply by bmallya** [reply]
This is what I had used earlier
`sources-exec,sources-file,sources-http,sources-internal_logs,sources-socket,sources-stdin,sources-syslog,sources-vector,sources-internal_metrics,sources-host_metrics,sinks-blackhole,sinks-console,sinks-file,sinks-http,sinks-socket,sinks-vector,transforms-add_fields,transforms-ansi_stripper,transforms-coercer,transforms-concat,transforms-dedupe,transforms-field_filter,transforms-filter,transforms-json_parser,transforms-key_value_parser,transforms-log_to_metric,transforms-logfmt_parser,transforms-merge,transforms-metric_to_log,transforms-reduce,transforms-regex_parser,transforms-remap,transforms-remove_fields,transforms-rename_fields,transforms-route,transforms-sample,transforms-split,transforms-tokenizer`

**Reply by bmallya** [reply]
I used the musl binaries from the https://packages.timber.io/vector/0.21.2/ and was able to get it work on the arm device. I don't have to compile it locally anymore.

**Reply by jszwedko** [reply]
Ah, great to hear!
